driver/terraform_provider.go

```diff
@@ -0,0 +1,84 @@
+package driver
+
+import "github.com/hashicorp/consul-terraform-sync/templates/hcltmpl"
+
+// TerraformProviderBlock contains provider arguments and environment variables
+// for the Terraform provider.
+type TerraformProviderBlock struct {
+	block hcltmpl.NamedBlock
```

Comment:
Suggestion (minor optional, not blocker) to name `block` to something else - like `arguments` mentioned in the comments. For me, at first read, `block` sounded broader/inclusive of `env` i.e. environment variables are declared in the block.  For me the block splits into environment and provider config information. I might be misunderstanding 'block' terminology though.

Reply:
Heh not "block"-er. To provide a bit more background on why I chose to keep block here. Block is an HCL terminology for an object

```hcl
object {
  arg = "value"
}
```

And in context of Terraform configuration, it would be the provider block which includes an HCL label to the block. I felt that renaming to arguments would imply only the body of the block and not the HCL label/provider name, which isn't the case.

```hcl
provider "foo" {
  arg = "value"
}
```

Block in both of these context refers to the syntax for a configuration file and I would believe is exclusive of the environment. However, since this adds CTS support for `task_env` as a meta-argument, which is not directly supported by the provider, the `task_env` is initially in the block when loading the user-CTS configs, which then gets removed before it gets written as TF config files.

Does block sound a bit more ok in context of HCL and Terraform syntax?

Reply:
Thanks for taking the time to share your thinking behind it! To your point about `task_env` initially being in the block, when I came across `NewTerraformProviderBlock`, I understood the relationship between the two, so it was more of an initial thing.